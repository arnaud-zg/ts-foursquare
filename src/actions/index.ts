import { TLifeAction } from './life'
import { TVenuesAction } from './venues'

export type TRootActionType = TVenuesAction | TLifeAction

export type TRootAction = TVenuesAction | TLifeAction

export * from './life'
export * from './venues'
