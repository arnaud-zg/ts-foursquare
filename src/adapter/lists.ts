import { NLists, NPayload } from '../../types'
import { adaptPayload } from './payload'
import { ECommonError } from '../constants/error'

export const adaptGetLists = (
  payload: NPayload.IPayload<NLists.IResponse>
): NLists.IList | null => {
  const response = adaptPayload<NLists.IResponse>(payload)

  if (!response || !response.list) {
    throw new Error(ECommonError.RESPONSE_NOT_VALID)
  }

  return response.list
}
