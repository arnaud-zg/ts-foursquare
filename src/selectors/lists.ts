import { NLists, NStore } from '../../types'

export const listSelector = (
  state: NStore.IState,
  listId: string
): NLists.IList => listsSelector(state).entities[listId]

export const listsSelector = (state: NStore.IState) => state.lists
