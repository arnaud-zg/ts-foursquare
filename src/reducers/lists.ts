import { createReducer } from 'typesafe-actions'
import { NListsState } from '../../types'
import { getListsAsync, TListsAction } from '../actions'

export const initialState: NListsState.IState = {
  entities: {},
}

export const listsReducer = createReducer<NListsState.IState, TListsAction>(
  initialState
).handleAction(
  getListsAsync.success,
  (state, action): NListsState.IState => {
    return {
      ...state,
      entities: {
        ...state.entities,
        [action.payload.id]: action.payload,
      },
    }
  }
)
