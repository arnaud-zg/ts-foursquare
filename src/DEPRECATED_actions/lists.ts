import { ActionType, createAsyncAction } from 'typesafe-actions'
import { NRequest } from '../../types'
import { adaptGetLists } from '../adapter'

export const getListsAsync = createAsyncAction(
  'GET_LISTS_REQUEST',
  'GET_LISTS_SUCCESS',
  'GET_LISTS_FAILURE',
  'GET_LISTS_CANCEL'
)<NRequest.IListsPayload, ReturnType<typeof adaptGetLists>, Error, string>()

export const listsActions = {
  getListsAsync,
}

export type TListsAction = ActionType<typeof listsActions>
