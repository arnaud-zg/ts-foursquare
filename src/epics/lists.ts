import { Epic } from 'redux-observable'
import { of } from 'rxjs/internal/observable/of'
import { catchError, filter, map, switchMap, takeUntil } from 'rxjs/operators'
import { isActionOf } from 'typesafe-actions'
import { NStore } from '../../types'
import { getListsAsync, TRootAction } from '../actions'
import { adaptGetLists } from '../adapter'
import { getObservableLists } from '../services'

export const getListsEpic: Epic<TRootAction, TRootAction, NStore.IState> = (
  action$,
  state$
) =>
  action$.pipe(
    filter(isActionOf(getListsAsync.request)),
    switchMap(action =>
      getObservableLists({ action, state$ }).pipe(
        map(adaptGetLists),
        map(getListsAsync.success),
        catchError(err => of(getListsAsync.failure(err))),
        takeUntil(action$.pipe(filter(isActionOf(getListsAsync.cancel))))
      )
    )
  )
