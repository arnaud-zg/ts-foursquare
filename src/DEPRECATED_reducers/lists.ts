import { createReducer } from 'typesafe-actions'
import { NListsState } from '../../types'
import { getListsAsync, TListsAction } from '../DEPRECATED_actions'

export const initialState: NListsState.IState = {
  entities: {},
}

export const listsReducer = createReducer<NListsState.IState, TListsAction>(
  initialState
).handleAction(
  getListsAsync.success,
  (state, action): NListsState.IState => {
    return {
      ...state,
      entities: {
        ...state.entities,
        [action.payload.id]: action.payload,
      },
    }
  }
)
