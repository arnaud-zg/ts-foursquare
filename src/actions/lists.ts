import { NRequest } from '../../types'
import { getObservableLists } from '../services/lists'
import { IStandaloneConfig } from '../standalone'

export interface GetListsProps {
  config?: IStandaloneConfig
  payload: NRequest.IListsPayload
}

export const getLists = ({ config, payload }: GetListsProps) =>
  getObservableLists({ config, payload }).toPromise()
