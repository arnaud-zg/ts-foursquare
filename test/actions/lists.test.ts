import { getListsAsync } from '../../src/actions/lists'

describe('actions/lists', () => {
  it(`should make a snapshot of getListsAsync`, () => {
    expect(getListsAsync).toMatchSnapshot()
  })
})
