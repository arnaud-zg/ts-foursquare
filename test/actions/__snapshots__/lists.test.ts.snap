// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions/lists should make a snapshot of getListsAsync 1`] = `
Object {
  "cancel": [Function],
  "failure": [Function],
  "request": [Function],
  "success": [Function],
}
`;
